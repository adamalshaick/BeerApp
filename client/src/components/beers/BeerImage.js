import React from "react";
import { Image, ImageWrapper, Thumbnail } from "../common/styles/Image";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";
import PropTypes from "prop-types";

const BeerImage = ({ showImage, beer, onClick }) => {
  const transitionOptions = { transitionName: "fade" };
  let image;
  if (showImage) {
    image = (
      <div key={beer}>
        <ImageWrapper onClick={onClick} />
        <Image
          src={beer.image_url}
          onError={e => {
            e.target.onerror = null;
            e.target.src =
              "https://www.equiinzert.cz/gallery/inzeraty/no-image.png";
          }}
          showImage={showImage}
        />
      </div>
    );
  }
  return (
    <>
      <Thumbnail
        showImage={showImage}
        src={beer.image_url}
        onError={e => {
          e.target.onerror = null;
          e.target.src =
            "https://www.equiinzert.cz/gallery/inzeraty/no-image.png";
        }}
        onClick={onClick}
      />

      <ReactCSSTransitionGroup {...transitionOptions}>
        {image}
      </ReactCSSTransitionGroup>
    </>
  );
};

BeerImage.propTypes = {
  showImage: PropTypes.bool.isRequired,
  beer: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};

export default BeerImage;
