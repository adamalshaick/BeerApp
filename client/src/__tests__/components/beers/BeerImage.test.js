import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import BeerImage from "../../../components/beers/BeerImage";
import {
  ImageWrapper,
  Image,
  Thumbnail
} from "../../../components/common/styles/Image";

const mockToggleImage = jest.fn();

describe("render component", () => {
  const Props = {
    beer: {},
    showImage: false
  };

  const wrapper = shallow(<BeerImage {...Props} onClick={mockToggleImage} />);
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("show the image", () => {
  const Props = {
    beer: {},
    showImage: true
  };
  const wrapper = shallow(<BeerImage {...Props} onClick={mockToggleImage} />);

  it("shows expanded image", () => {
    expect(wrapper.find(ImageWrapper).length).toEqual(1);
    expect(wrapper.find(Image).length).toEqual(1);
  });

  it("calls toggle function after clicking on image wrapper", () => {
    wrapper.find(ImageWrapper).simulate("click");
    expect(mockToggleImage.mock.calls.length).toEqual(1);
  });
});

describe("dont show the image", () => {
  const Props = {
    beer: {},
    showImage: false
  };
  const wrapper = shallow(<BeerImage {...Props} onClick={mockToggleImage} />);

  it("doesn't show expanded image", () => {
    expect(wrapper.find(ImageWrapper).length).toEqual(0);
    expect(wrapper.find(Image).length).toEqual(0);
  });

  it("calls toggle function after clicking on thumbnail", () => {
    wrapper.find(Thumbnail).simulate("click");
    expect(mockToggleImage.mock.calls.length).toEqual(2);
  });
});
