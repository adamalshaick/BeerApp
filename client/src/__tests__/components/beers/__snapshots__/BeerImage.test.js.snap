// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<Fragment>
  <styled.img
    onClick={[MockFunction]}
    onError={[Function]}
    showImage={false}
  />
  <CSSTransitionGroup
    transitionAppear={false}
    transitionEnter={true}
    transitionLeave={true}
    transitionName="fade"
  />
</Fragment>
`;
